server/server.go

```diff
@@ -306,9 +310,10 @@ func (s *server) invokeHandleEndpoint() endpoint.Endpoint {
 		ri := rpcinfo.GetRPCInfo(ctx)
 		methodName := ri.Invocation().MethodName()
 		serviceName := ri.Invocation().ServiceName()
-		svc := s.svcs.getService(serviceName)
+		svc := s.svcs.svcMap[serviceName]
 		if svc == nil {
-			svc = s.svcs.defaultSvc
+			// NOTE: svc should always be non-nil here
+			svc = s.svcs.svcSearchMap[methodName]
```

Comment:
Is it really necessary to have a fallback check? The method name has already been searched once, so why is there a possibility of not finding the service here?

Reply:
@jayantxie You are right. Thank you.